sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/LookupCatalog.scala

```diff
@@ -103,16 +103,7 @@ private[sql] trait LookupCatalog extends Logging {
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Identifier)] = {
       assert(nameParts.nonEmpty)
       if (nameParts.length == 1) {
-        // If the current catalog is session catalog, the current namespace is not used because
-        // the single-part name could be referencing a temp view, which doesn't belong to any
-        // namespaces. An empty namespace will be resolved inside the session catalog
-        // implementation when a relation is looked up.
-        val ns = if (CatalogV2Util.isSessionCatalog(currentCatalog)) {
-          Array.empty[String]
-        } else {
-          catalogManager.currentNamespace
-        }
-        Some((currentCatalog, Identifier.of(ns, nameParts.head)))
+        Some((currentCatalog, Identifier.of(catalogManager.currentNamespace, nameParts.head)))
```

Comment:
Hmm, this adds current namespace to the returned identifier? So we cannot tell if a returned identifier is for a temp view?

Reply:
I think `ResolveTempViews` needs to be updated to look up twice (one with namespace and one without).

Reply:
if you look at `SessionCatalog.setCurrentDatabase`, we forbid to set `global_temp` as the current database.

This is to make the resolution of temp view simple: we look up temp view via the name parts given by users literally. No name extension is allowed.

The general rule of relation resolution is: we try to look up temp view first, then tables/permanent views. When we call `CatalogAndIdentifier`, the temp view lookup should already be tried, or the caller side don't want to resolve temp views.

Reply:
> When we call `CatalogAndIdentifier`, the temp view lookup should already be tried, or the caller side don't want to resolve temp views.

So, `ResolveCatalogs` should be applied after `ResolveTables` and `ResolveRelations`?

Reply:
these 2 are different frameworks so I don't see a way to guarantee the order. I think what we should do is to migrate to the new framework.